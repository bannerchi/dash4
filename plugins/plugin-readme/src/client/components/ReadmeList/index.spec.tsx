/// <reference types="@types/jest" />
const useFullScreenMock = jest.fn().mockImplementation(() => ({
	fullscreen: false,
	enableFullscreen: jest.fn(),
	disableFullscreen: jest.fn(),
}));
jest.mock('../../hooks', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	useMarkdownData: jest.fn().mockImplementation(() => '<div>markdown</div>'),
	useFullscreen: useFullScreenMock,
}));

import { render } from '@testing-library/react';
import React from 'react';
import { getMatchingQuery, Readme, ReadmeList } from '.';

describe('PluginReadmeList', () => {
	describe('ReadmeList', () => {
		test('exist', () => {
			expect(typeof ReadmeList).toBe('object');
		});
		test('render default', () => {
			const { container } = render(
				<ReadmeList
					id={'i1-root'}
					name={'PluginReadmeList'}
					lowerCaseName={'plugin-readme-list'}
					clientConfig={{
						files: [
							{
								title: 'title',
								file: '/path-to-file',
								id: 'id',
							},
						],
					}}
				/>
			);
			expect(container.firstChild).toMatchSnapshot();
		});
		test('render fullscreen', () => {
			useFullScreenMock.mockReturnValue({
				fullscreen: true,
				enableFullscreen: jest.fn(),
				disableFullscreen: jest.fn(),
			});
			const { container } = render(
				<ReadmeList
					id={'i1-root'}
					name={'PluginReadmeList'}
					lowerCaseName={'plugin-readme-list'}
					clientConfig={{
						title: 'custom title',
						files: [
							{
								title: 'title',
								file: '/path-to-file',
								id: 'id',
							},
						],
					}}
				/>
			);
			expect(container.firstChild).toMatchSnapshot();
			jest.restoreAllMocks();
		});
	});

	describe('Readme', () => {
		test('exist', () => {
			expect(typeof Readme).toBe('object');
		});
	});

	describe('getMatchingQuery', () => {
		test('first one matches', () => {
			expect(
				getMatchingQuery({
					1: true,
					2: false,
					3: false,
					4: false,
					5: false,
					6: false,
				})
			).toBe('1');
		});
		test('last one matches', () => {
			expect(
				getMatchingQuery({
					1: false,
					2: false,
					3: false,
					4: false,
					5: false,
					6: true,
				})
			).toBe('6');
		});
		test('multiple matches sould return first one', () => {
			expect(
				getMatchingQuery({
					1: false,
					2: true,
					3: false,
					4: false,
					5: false,
					6: true,
				})
			).toBe('2');
		});
	});
});
