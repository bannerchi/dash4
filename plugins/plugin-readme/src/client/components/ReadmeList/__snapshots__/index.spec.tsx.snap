// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginReadmeList ReadmeList render default 1`] = `
<div
  class="Component-window-0-1-11 Component-windowBright-0-1-13 Component-windowAutoStretch-0-1-12"
  tabindex="1"
>
  <div
    class="ReadmeListRaw-windowBody-0-1-1 Component-body-0-1-16"
  >
    <ul
      class="ReadmeListRaw-ul-0-1-3"
    >
      <li
        class="ReadmeListRaw-li-0-1-4 undefined"
      >
        <button
          class="Component-box-0-1-17"
        >
          <div
            aria-hidden="true"
            class="Component-markdownWindow-0-1-19"
          >
            <div
              class="Component-markdownSpacing-0-1-20"
            >
              <div
                class="Component-markdown-0-1-21"
              >
                <div>
                  markdown
                </div>
              </div>
            </div>
          </div>
          <h6
            class="Component-title-0-1-18"
          >
            title
          </h6>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PluginReadmeList ReadmeList render fullscreen 1`] = `
<div
  class="Component-window-0-1-11 Component-windowBright-0-1-13 Component-windowAutoStretch-0-1-12"
  tabindex="1"
>
  <div
    class=" Component-header-0-1-26"
  >
    <p
      class="Component-title-0-1-28"
    >
      custom title
    </p>
  </div>
  <div
    class="ReadmeListRaw-windowBody-0-1-1 Component-body-0-1-16"
  >
    <ul
      class="ReadmeListRaw-ul-0-1-3"
    >
      <li
        class="ReadmeListRaw-li-0-1-4 undefined"
      >
        <button
          class="Component-box-0-1-17"
        >
          <div
            aria-hidden="true"
            class="Component-markdownWindow-0-1-19"
          >
            <div
              class="Component-markdownSpacing-0-1-20"
            >
              <div
                class="Component-markdown-0-1-21"
              >
                <div>
                  markdown
                </div>
              </div>
            </div>
          </div>
          <h6
            class="Component-title-0-1-18"
          >
            title
          </h6>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
